owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt

```diff
@@ -0,0 +1,203 @@
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+
+package com.owncloud.android.lib.resources.shares
+
+import android.os.Parcel
+import android.os.Parcelable
+import com.owncloud.android.lib.common.utils.Log_OC
+import com.owncloud.android.lib.resources.files.FileUtils
+import java.io.Serializable
+
+/**
+ * Contains the data of a Share from the Share API
+ *
+ * @author masensio
+ * @author David A. Velasco
+ * @author David González Verdugo
+ */
+class RemoteShare : Parcelable, Serializable {
+    var shareWith: String = ""
+    var path: String = ""
+    var token: String = ""
+    var sharedWithDisplayName: String = ""
+    var sharedWithAdditionalInfo: String = ""
+    var name: String = ""
+    var shareLink: String = ""
+    var fileSource: Long = 0
+    var itemSource: Long = 0
+    var shareType: ShareType? = null
+    var permissions: Int = 0
+    var sharedDate: Long = 0
+    var expirationDate: Long = 0
+    var isFolder: Boolean = false
+    var userId: Long = 0
+    var remoteId: Long = 0
+        private set
+
+    val isPasswordProtected: Boolean
+        get() = ShareType.PUBLIC_LINK == shareType && shareWith.isNotEmpty()
+
+    constructor() : super() {
+        resetData()
+    }
+
+    constructor(path: String?) {
+        resetData()
+        if (path == null || path.isEmpty() || !path.startsWith(FileUtils.PATH_SEPARATOR)) {
+            Log_OC.e(TAG, "Trying to create a RemoteShare with a non valid path")
+            throw IllegalArgumentException("Trying to create a RemoteShare with a non valid path: " + path!!)
+        }
+        this.path = path
+    }
+
+    /**
+     * Used internally. Reset all file properties
+     */
+    private fun resetData() {
+        shareWith = ""
+        path = ""
+        token = ""
+        sharedWithDisplayName = ""
+        sharedWithAdditionalInfo = ""
+        name = ""
+        shareLink = ""
+        fileSource = 0
+        itemSource = 0
+        shareType = ShareType.NO_SHARED
+        permissions = DEFAULT_PERMISSION
+        sharedDate = 0
+        expirationDate = 0
+        sharedWithAdditionalInfo = ""
+        isFolder = false
+        userId = -1
```

Comment:
why don't you use constants for initial values?

Reply:
This is a question that should have been done in a code review 5 years ago: 

https://github.com/owncloud/android-library/blame/87a8b418ded714e64aa77c9a007801b6cf4a985e/owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/OCShare.java#L132

Not sure if is going to be possible (in terms of time) to improve the new code added by me and legacy code (converted from Java to Kotlin). Being strict, just the new code should be reviewed and changed in code reviews.

Otherwise...

![17201963](https://user-images.githubusercontent.com/15261363/53824254-d3e06c00-3f73-11e9-9b6a-e089c5093e52.jpg)